src/emu/inpttype.h

```diff
@@ -851,6 +851,8 @@ inline void construct_core_types_UI(simple_list<input_type_entry> &typelist)
 	INPUT_PORT_DIGITAL_TYPE( 0, UI,      UI_FOCUS_PREV,       "UI Focus Previous",      input_seq(KEYCODE_TAB, KEYCODE_LSHIFT) )
 	INPUT_PORT_DIGITAL_TYPE( 0, UI,      UI_SELECT,           "UI Select",              input_seq(KEYCODE_ENTER, input_seq::or_code, JOYCODE_BUTTON1_INDEXED(0), input_seq::or_code, KEYCODE_ENTER_PAD) )
 	INPUT_PORT_DIGITAL_TYPE( 0, UI,      UI_CANCEL,           "UI Cancel",              input_seq(KEYCODE_ESC) )
+	INPUT_PORT_DIGITAL_TYPE( 0, UI,      UI_QUIT,             "UI Quit",                input_seq(KEYCODE_ESC, input_seq::not_code, KEYCODE_LSHIFT) )
+	INPUT_PORT_DIGITAL_TYPE( 0, UI,      UI_FORCE_QUIT,       "UI Force Quit",          input_seq(KEYCODE_ESC, KEYCODE_LSHIFT) )
```

Comment:
I don’t like the choice of name.  “Force quit” implies that it prevents the system from preventing the quit from happening.  With this implementation, there’s no way for a device to cancel/defer the exit anyway, so it’s always forced.  It’s actually “quit without warning and without saving NVRAM contents”.

Reply:
Could you suggest alternate names for the input? I agree that its present nomenclature is a bit dubious, as are the finer aspects of its technical implementation.

Reply:
Read the overall comments at the top.  This input is something more along the lines of "abandon state" which requires a lot more thought to implement properly.

Reply:
I don't take that as a meaningful answer, though I'm not in any hurry to merge this PR now and may leave out the "Force Quit" part when I revise it later.